### PR TITLE
feature: Add support for running OpenQASM programs

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*
feature: Add support for running OpenQASM programs

*Testing done:*
Team tested.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md) doc
- [ x ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ x ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-schemas-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ x ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,205 +1,199 @@
 # Changelog
```
Comment (jheckey) [inline CHANGELOG.md:2]: Lost 1.7.2 note; need a 1.8.0 note

File: src/braket/_schemas/_version.py [hunk 2]
```diff
@@ -15,4 +15,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "1.7.3.dev0"
+__version__ = "1.8.0.dev0"
```
Comment (jheckey) [inline src/braket/_schemas/_version.py:18]: Version is 1.8.0.

File: src/braket/_schemas/_version.py [hunk 3]
```diff
@@ -15,4 +15,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "1.7.3.dev0"
+__version__ = "1.7.2"
```
Comment (kshitijc) [inline src/braket/_schemas/_version.py:18]: These are handled by the release pipeline. Why is this being modified manually ?

**Review by jheckey — DISMISSED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (math411) [reply CHANGELOG.md:2]: Added 1.7.2 note

File: src/braket/_schemas/_version.py (hunk 2 above)
Comment (math411) [reply src/braket/_schemas/_version.py]: Changed to 1.7.2

File: src/braket/_schemas/_version.py (hunk 3 above)
Comment (math411) [reply src/braket/_schemas/_version.py:18]: Changing to the last release in pypi

Comment (math411) [reply src/braket/_schemas/_version.py]: Rolled back the changed to 1.7.3.dev0

**Review by jheckey — APPROVED** [review]

**Pull request merged**